packages/addon/lua/gelly/mods/blood-mod/init.lua

```diff
@@ -96,21 +128,40 @@ end
 hook.Add(
 	"GellyXDamage",
 	"gelly.builtin.blood-mod",
-	function(_, attacker, position, force, damage, type)
+	function(victim, attacker, position, force, damage, type)
 		if
-			not attacker:IsPlayer()
-			and not attacker:IsNPC()
-			and attacker:GetClass() ~= "prop_ragdoll"
+			not victim:IsValid() or
+			not victim:IsPlayer() and
+			not victim:IsNPC() and
+			not victim:IsRagdoll()
```

Comment:
I'm not certain that brackets are absolutely required around the `and` conditions here with Lua's short circuit evaluation, but I'd like them for readability regardless